File: C++/super-washing-machines.cpp
// Time:  O(n)
// Space: O(1)

class Solution {
public:
    int findMinMoves(vector<int>& machines) {
        int sum = accumulate(machines.begin(), machines.end(), 0);
        if (sum % machines.size() != 0) {
            return -1;
        }
        
        int result = 0, target = sum / machines.size(), curr = 0;
        for (auto i = 0; i < machines.size(); ++i) {
            curr += machines[i] - target;
            result = max(result, max(machines[i] - target, abs(curr)));
        }
        return result;
    }
};

File: Python/super-washing-machines.py
# Time:  O(n)
# Space: O(1)

class Solution(object):
    def findMinMoves(self, machines):
        """
        :type machines: List[int]
        :rtype: int
        """
        total = sum(machines)
        if total % len(machines): return -1

        result, target, curr = 0, total / len(machines), 0
        for n in machines:
            curr += n - target
            result = max(result, max(n - target, abs(curr)))
        return result

